test/L1VotePool.t.sol

```diff
@@ -23,13 +23,15 @@ contract L1VotePoolHarness is L1VotePool, Test {
   function receiveWormholeMessages(
     bytes memory payload,
     bytes[] memory additionalVaas,
-    bytes32 callerAddr,
+    bytes32 sourceAddress,
     uint16 sourceChain,
     bytes32 deliveryHash
-  ) public override onlyRelayer {
+  ) public override onlyRelayer isRegisteredSender(sourceChain, sourceAddress) {
```

Comment:
Good question, and this function is not ideal. The issue here is we need the proposal to be active to make a successful cross chain call, and we use Wormhole's testing SDK to do this, specifically a function called `performRelay`. This function doesn't give us an opportunity to insert custom logic before the cross-chain call. 

Some things we have done and can do:

1. I added a comment specfying a reason in a upstack pr.
2. Make an issue on Wormholes's testing repo. They haven't been responsive on other issues I have opened so I don't have high hopes they would address this right away
3. We can fork and patch their testing utilities
4. Document and leave as is for now 

The test that needs this setup can be found [here](https://github.com/ScopeLift/l2-flexible-voting/blob/main/test/L1VotePool.t.sol#L140). It is more of an integration rather than a unit test.

Reply:
The above all makes sense

> This function doesn't give us an opportunity to insert custom logic before the cross-chain call.

But just to clarify, I'd have expected that custom logic to be inserted before calling the parent, e.g.:

```solidity
contract L1VotePoolHarness is L1VotePool, Test {
  function receiveWormholeMessages(...) external {
    (uint256 proposalId,,,) = abi.decode(payload, (uint256, uint128, uint128, uint128));
    _jumpToActiveProposal(proposalId);

    // Above we executed custom logic, now we call the parent method.
    // Instead we currently call an internal method here
    L1VotePool.receiveWormholeMessages(...);
  }
```

Maybe the method in `L1VotePoolHarness` should be called `exposed_receiveCastVoteWormholeMessages` instead? i.e. Is it actually testing `_receiveCastVoteWormholeMessages` and not `receiveWormholeMessages`?

Reply:
This is a good clarification. I think you are right we should rename the internal method to `receiveWormholeMessages`, and then when it is called in the `L1ERC20Bridge` use `super`.

And I do think you could make an argument this is testing `receiveWormholeMessages` instead of `_receiveCastVoteWormholeMessages`. We could get away with not using `performDelivery`, and rename the harness methods to use the `exposed` naming convention.

Reply:
https://github.com/ScopeLift/l2-flexible-voting/issues/51